### PR TITLE
Always honor `Strict Resolution` (Fixes #15432)

### LINKED ISSUE #15432: `typeclasses eauto with ..` is unaffected by `Strict Resolution`.
#### Description of the problem

The code below shows inconsistencies in the behavior of different entry points to typeclass search w.r.t. `Strict Resolution`. I expect all of the queries to fail, yet the last one succeeds.
```coq
#[local] Set Typeclasses Strict Resolution.
Class C (P : Prop) (out : Prop) := { c : out -> P}.
#[refine] Instance : C True False := { c := _ }. abstract auto. Defined.
Goal exists q, C True q.
  eexists.
  Fail apply _.
  Fail typeclasses eauto.
  typeclasses eauto with typeclass_instances.
```

#### Coq Version
8.13.1, 8.14.0

### PR DESCRIPTION
**Pull request by Janno**
Fixes / closes #15432

- [x] Added / updated **test-suite**.

- [x] Added **changelog**.

<!-- Pointers to relevant developer documentation:

Contributing guide: https://github.com/coq/coq/blob/master/CONTRIBUTING.md

Test-suite: https://github.com/coq/coq/blob/master/test-suite/README.md

Changelog: https://github.com/coq/coq/blob/master/doc/changelog/README.md

Building the doc: https://github.com/coq/coq/blob/master/doc/README.md
Sphinx: https://github.com/coq/coq/blob/master/doc/sphinx/README.rst
doc_gram: https://github.com/coq/coq/blob/master/doc/tools/docgram/README.md

Overlays: https://github.com/coq/coq/blob/master/dev/ci/user-overlays/README.md

### CODE REVIEW

File: tactics/class_tactics.ml [hunk 1]
```diff
@@ -258,7 +258,7 @@ and e_my_find_search db_list local_db secvars hdc complete only_classes env sigm
   let allowed_evars =
     let all = Evarsolve.AllowedEvars.all in
     match hdc with
-    | Some (hd,_) when only_classes ->
```
Comment (SkySkimmer) [inline tactics/class_tactics.ml:261]: I think this is the only use of `only_classes` so the argument should be removed.

Comment (Janno) [reply tactics/class_tactics.ml:261]: No, it's used further down. I don't think dune would even let me compile a function with an unused argument..

Comment (Janno) [reply tactics/class_tactics.ml:261]: Indeed, it's not really used. I have a fixed version but I'll let CI finish before I push.

Comment (Janno) [reply tactics/class_tactics.ml:261]: CI finished successfully. I pushed the removal of the argument.

**Comment by Janno** [comment]
Nobody seems to be relying on this undocumented behavior. Anything else that needs doing here or can this go in?

**Review by ppedrot — APPROVED** [review]
The expected behaviour of typeclasses eauto (as compared to vanilla TC resolution) is not obvious to me but the discrepancy depending on the use of a db is clearly a bug. The fix seems fine, so AFAICT this LGTM.

**Comment by ppedrot** [comment]
@coqbot run full ci

**Comment by ppedrot** [comment]
@Janno you may want to add a changelog, but this issue is really a fringe case so it's up to you.

**Comment by Janno** [comment]
Nobody else seems to have noticed it so the value of a changelog entry is questionable. I'd skip it.

**Comment by Janno** [comment]
I just realized that pushing the changelog entry stopped CI.

@coqbot run full ci

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**